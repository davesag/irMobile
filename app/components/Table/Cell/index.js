import React from 'react'
import PropTypes from 'prop-types'
import { Text } from 'react-native-paper'

import styles from '../styles'

const Cell = ({ value, numeric, style }) =>
  numeric ? (
    <Text style={[styles.data, style]}>{value}</Text>
  ) : (
    <Text style={style}>{value}</Text>
  )

Cell.propTypes = {
  value: PropTypes.string.isRequired,
  numeric: PropTypes.bool,
  style: PropTypes.object
}

Cell.defaultProps = {
  numeric: false,
  style: {}
}

export default Cell
