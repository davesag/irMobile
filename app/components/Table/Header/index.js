import React from 'react'
import PropTypes from 'prop-types'
import { View } from 'react-native'
import { Text } from 'react-native-elements'

import styles from '../styles'

const Header = ({ data }) =>
  data ? (
    <View style={styles.row}>
      {data.map(text => (
        <View key={text} style={styles.headerCell}>
          <Text style={styles.header}>{text}</Text>
        </View>
      ))}
    </View>
  ) : null

Header.propTypes = {
  data: PropTypes.arrayOf(PropTypes.string)
}

Header.defaultProps = {
  data: null
}

export default Header
