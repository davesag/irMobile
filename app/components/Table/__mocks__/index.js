import React from 'react'
import { View } from 'react-native'

export default jest.fn(props => <View>Table</View>)
