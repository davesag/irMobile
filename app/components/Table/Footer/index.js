import React from 'react'
import PropTypes from 'prop-types'
import { DataTable, withTheme } from 'react-native-paper'

import Cell from '../Cell'

import styles from '../styles'

const Footer = ({ data, theme, makeKey }) =>
  data ? (
    <DataTable.Row style={styles.footerStyle(theme)}>
      {data.map((value, i) => {
        const align = i === data.length - 1 ? 'right' : undefined

        return (
          <DataTable.Title
            key={makeKey(value, i)}
            style={styles.cellStyle(i, data, theme, align)}
          >
            <Cell value={value} numeric={i !== 0} />
          </DataTable.Title>
        )
      })}
    </DataTable.Row>
  ) : null

Footer.propTypes = {
  data: PropTypes.arrayOf(PropTypes.string),
  theme: PropTypes.object.isRequired,
  makeKey: PropTypes.func.isRequired
}

Footer.defaultProps = {
  data: null
}

export default withTheme(Footer)
