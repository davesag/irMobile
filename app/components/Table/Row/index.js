import React from 'react'
import PropTypes from 'prop-types'
import { DataTable, withTheme } from 'react-native-paper'

import Cell from '../Cell'

import styles from '../styles'

const Row = ({ data, theme, makeKey }) =>
  data ? (
    <DataTable.Row style={styles.rowStyle(theme)}>
      {data.map((value, i) => (
        <DataTable.Cell
          style={styles.cellStyle(i, data, theme)}
          numeric={i !== 0}
          key={makeKey(value, i)}
        >
          <Cell value={value} numeric={i !== 0} />
        </DataTable.Cell>
      ))}
    </DataTable.Row>
  ) : null

Row.propTypes = {
  data: PropTypes.arrayOf(PropTypes.string).isRequired,
  theme: PropTypes.object.isRequired,
  makeKey: PropTypes.func.isRequired
}

export default withTheme(Row)
