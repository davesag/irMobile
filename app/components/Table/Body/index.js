import React from 'react'
import PropTypes from 'prop-types'
import { ScrollView } from 'react-native'

import Row from '../Row'

const Body = ({ data, makeKey }) => (
  <ScrollView>
    {data.map(row => (
      <Row
        key={`row-${row[0]}`}
        data={row}
        makeKey={(value, i) => makeKey(row[0], i)}
      />
    ))}
  </ScrollView>
)

Body.propTypes = {
  data: PropTypes.arrayOf(PropTypes.arrayOf(PropTypes.string)).isRequired,
  makeKey: PropTypes.func.isRequired
}

Body.defaultProps = {
  data: null
}

export default Body
