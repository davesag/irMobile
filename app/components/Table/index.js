import React from 'react'
import PropTypes from 'prop-types'
import { View } from 'react-native'

import Header from './Header'
import Footer from './Footer'
import Row from './Row'

import styles from './styles'

const Table = ({
  body,
  header, // null or array of strings
  footer, // null or array of strings
  tableStyle
}) => {
  return (
    <View style={[styles.table, tableStyle]}>
      <Header data={header} />
      {body.map(row => (
        <Row data={row} key={row[0]} />
      ))}
      <Footer data={footer} />
    </View>
  )
}

Table.propTypes = {
  body: PropTypes.arrayOf(PropTypes.arrayOf(PropTypes.string)).isRequired,
  header: PropTypes.arrayOf(PropTypes.string),
  footer: PropTypes.arrayOf(PropTypes.string),
  tableStyle: PropTypes.object
}

Table.defaultProps = {
  tableStyle: null,
  header: null,
  footer: null
}

export default Table
