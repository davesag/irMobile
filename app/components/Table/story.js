import React from 'react'
import { storiesOf } from '@storybook/react-native'
import { SafeAreaView } from 'react-native'

import Table from '.'

const body = [['A', 'B', 'C', 'D'], ['E', 'F', 'G', 'H']]

const header = ['Aye', 'Bee', 'Sea', 'Dee']
const footer = ['Total', '', '', 'Eee']

storiesOf('Table', module)
  .add('simple', () => (
    <SafeAreaView>
      <Table body={body} />
    </SafeAreaView>
  ))
  .add('with header', () => (
    <SafeAreaView>
      <Table header={header} body={body} />
    </SafeAreaView>
  ))
  .add('with footer', () => (
    <SafeAreaView>
      <Table footer={footer} body={body} />
    </SafeAreaView>
  ))
  .add('with header, footer, and custom style', () => (
    <SafeAreaView>
      <Table
        header={header}
        footer={footer}
        body={body}
        tableStyle={{ padding: 10 }}
      />
    </SafeAreaView>
  ))
