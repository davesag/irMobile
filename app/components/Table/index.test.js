import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import theme from '../../theme'

import Table from '.'

const body = [['A', 'B', 'C'], ['D', 'E', 'F']]
const header = ['Aye', 'Bee', 'Sea']
const footer = ['Total', '', 'Dee']

const customStyle = { padding: 10 }

let tree

jest.mock('./Header')
jest.mock('./Body')
jest.mock('./Footer')

describe('rendering', () => {
  describe('without header or footer', () => {
    beforeAll(() => {
      tree = renderer.create(<Table body={body} theme={theme} />)
    })

    it('rendered correctly ', () => {
      expect(tree).toMatchSnapshot()
    })
  })

  describe('with header', () => {
    beforeAll(() => {
      tree = renderer.create(
        <Table body={body} header={header} theme={theme} />
      )
    })

    it('rendered correctly ', () => {
      expect(tree).toMatchSnapshot()
    })
  })

  describe('with footer', () => {
    beforeAll(() => {
      tree = renderer.create(
        <Table body={body} footer={footer} theme={theme} />
      )
    })

    it('rendered correctly ', () => {
      expect(tree).toMatchSnapshot()
    })
  })

  describe('with header and footer and custom style', () => {
    beforeAll(() => {
      tree = renderer.create(
        <Table
          body={body}
          header={header}
          footer={footer}
          tableStyle={customStyle}
          theme={theme}
        />
      )
    })

    it('rendered correctly ', () => {
      expect(tree).toMatchSnapshot()
    })
  })
})
