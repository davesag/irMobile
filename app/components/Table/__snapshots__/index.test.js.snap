// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering with footer rendered correctly  1`] = `
<View>
  <View>
    <View>
      <Text>
        A
      </Text>
      <Text>
        B
      </Text>
      <Text>
        C
      </Text>
    </View>
    <View>
      <Text>
        D
      </Text>
      <Text>
        E
      </Text>
      <Text>
        F
      </Text>
    </View>
  </View>
  <View>
    <Text>
      d
    </Text>
    <Text>
      d
    </Text>
    <Text>
      d
    </Text>
  </View>
</View>
`;

exports[`rendering with header and footer and custom style rendered correctly  1`] = `
<View>
  <View>
    <Text>
      d
    </Text>
    <Text>
      d
    </Text>
    <Text>
      d
    </Text>
  </View>
  <View>
    <View>
      <Text>
        A
      </Text>
      <Text>
        B
      </Text>
      <Text>
        C
      </Text>
    </View>
    <View>
      <Text>
        D
      </Text>
      <Text>
        E
      </Text>
      <Text>
        F
      </Text>
    </View>
  </View>
  <View>
    <Text>
      d
    </Text>
    <Text>
      d
    </Text>
    <Text>
      d
    </Text>
  </View>
</View>
`;

exports[`rendering with header rendered correctly  1`] = `
<View>
  <View>
    <Text>
      d
    </Text>
    <Text>
      d
    </Text>
    <Text>
      d
    </Text>
  </View>
  <View>
    <View>
      <Text>
        A
      </Text>
      <Text>
        B
      </Text>
      <Text>
        C
      </Text>
    </View>
    <View>
      <Text>
        D
      </Text>
      <Text>
        E
      </Text>
      <Text>
        F
      </Text>
    </View>
  </View>
</View>
`;

exports[`rendering without header or footer rendered correctly  1`] = `
<View>
  <View>
    <View>
      <Text>
        A
      </Text>
      <Text>
        B
      </Text>
      <Text>
        C
      </Text>
    </View>
    <View>
      <Text>
        D
      </Text>
      <Text>
        E
      </Text>
      <Text>
        F
      </Text>
    </View>
  </View>
</View>
`;
